### PR TITLE
Enable varchar equality predicate pushdown for ClickHouse connector

### PR DESCRIPTION
**Pull request by sylph-eu**
## Description
This PR depends partially addresses https://github.com/trinodb/trino/issues/7100 (only textual types for now).

Type mapping unit tests were extended with more cases. In addition to `Type` and `Nullable(Type)` I'm testing whether `LowCardinality` modifier causes issues. In the past ClickHouse had some incompatibilities between regular and LowCardinality types, therefore it's worth putting them in the unit tests.

Only equality predicate is enabled for now as ClickHouse does not support collation. Respective unit tests were put into `TestClickHouseConnectorTest.testTextualPredicatePushdown`.

## Additional context and related issues

I hope to continue working on https://github.com/trinodb/trino/issues/7100 to support pushdown for more more types (enums, decimals, etc).

## Release notes

(X) Release notes are required, with the following suggested text:

```markdown
# ClickHouse
* Improve performance when pushing down equality predicate on textual types. ({issue}`7100`)
```

### CODE REVIEW

**Comment by sylph-eu** [comment]
Rebased.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java:70]: nit: Sort alphabetically.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:454]: Could you add a helper method instead of repeating the same roundtrip?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java]: Done, thanks.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java]: Also done.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:442]: No need to take SqlDataTypeTest as the argument. See `TestPostgreSqlTypeMapping#arrayDecimalTest` as the example. 

Also, please avoid abbreviations. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-abbreviations

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:449]: Fix wrong indentation.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:502]: These new tests are unrelated to "Enable varchar equality predicate pushdown for ClickHouse connector", right? Please extract the changes as a preparatory commit.

Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:502]: I copied these checks from respective postgres unit tests, to bring them to some kind of sync with regard to varchar/string support. So they're related to the PR. 

I can drop them if you think they're excessive.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java]: Addressed.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java]: Addressed, thanks.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:502]: Please separate a commit.

Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:502]: Sorry for stupid question, what do you mean by "preparatory commit"? A separate PR with new unit tests only? Or a separate commit within this PR?

Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:502]: I mean the latter, "a separate commit within this PR".

Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:502]: Done, please see https://github.com/trinodb/trino/pull/21557/commits/c15a930fa638604e4bf0bfe55ba8435ef2ced9d2

**Comment by sylph-eu** [comment]
Rebased

**Comment by nileshsinha1993** [comment]
Hi @sylph-eu when are we planning to release this PR?

**Comment by sylph-eu** [comment]
> Hi @sylph-eu when are we planning to release this PR? Trino version.

I'm not in control of merging/releasing this PR, I'm a side contributor. It's waiting for @ebyhr to complete the review and merge it. This, in turn, will unblock subsequent ones (please see #7100 for the current list).

**Comment by sylph-eu** [comment]
Rebased once more.

**Comment by bhargav2427** [comment]
Hello @ebyhr, this is very good feature and our team is eagerly waiting for it. It will be really good if you can check.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:442]: Change to static and move to after `testTrinoVarchar`. 

Also, please follow the commit message guideline: https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:478]: This change is unrelated to "Enable varchar equality predicate pushdown for ClickHouse-connector". Please separate a commit.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:531]: This change should belong to "extra unit tests" commit.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java:445]: Moved to the unit test commit.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java
Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java]: Done.

Comment (sylph-eu) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java]: Renamed commit to `Extra unit tests for type mapping`

**Review by ebyhr — APPROVED** [review]

**Pull request merged**
